core/controllers/cron_test.py

```diff
@@ -316,6 +321,149 @@ def _mock_logging_function(msg, *args):
+
+    def setUp(self):
+        super(
+            CronMailContributorDashboardReviewerOpportunitiesHandlerTests,
+            self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.signup(self.AUTHOR_EMAIL, self.AUTHOR_USERNAME)
+        self.author_id = self.get_user_id_from_email(self.AUTHOR_EMAIL)
+        self.signup(self.REVIEWER_EMAIL, self.REVIEWER_USERNAME)
+        self.reviewer_id = self.get_user_id_from_email(self.REVIEWER_EMAIL)
+        user_services.update_email_preferences(
+            self.reviewer_id, True, False, False, False)
+        self.save_new_valid_exploration(self.target_id, self.author_id)
+
+        self.can_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+        self.cannot_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', False)
+        self.can_send_reviewer_emails = self.swap(
+            config_domain, 'NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_IS_ENABLED',
+            True)
+        self.cannot_send_reviewer_emails = self.swap(
+            config_domain, 'NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_IS_ENABLED',
+            False)
+        self.testapp_swap = self.swap(
+            self, 'testapp', webtest.TestApp(main_cron.app))
+
+        self.reviewers_suggestion_email_infos = []
+        self.reviewer_ids = []
+
+    def test_email_not_sent_if_sending_reviewer_emails_is_not_enabled(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        with self.cannot_send_reviewer_emails, self.testapp_swap:
+            self.get_html_response(
+                '/cron/mail/contributor_dashboard_reviewers/'
+                'review_opportunities')
+
+        self.assertEqual(len(self.reviewer_ids), 0)
+        self.assertEqual(len(self.reviewers_suggestion_email_infos), 0)
+
+        self.logout()
+
+    def test_email_not_sent_if_sending_emails_is_not_enabled(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        with self.cannot_send_emails, self.testapp_swap:
+            self.get_html_response(
+                '/cron/mail/contributor_dashboard_reviewers/'
+                'review_opportunities')
+
+        self.assertEqual(len(self.reviewer_ids), 0)
+        self.assertEqual(len(self.reviewers_suggestion_email_infos), 0)
+
+        self.logout()
+
+    def test_email_sent_to_reviewer_if_sending_reviewer_emails_is_enabled(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        user_services.allow_user_to_review_translation_in_language(
+            self.reviewer_id, self.language_code)
+        translation_suggestion = self._create_translation_suggestion()
+        expected_reviewable_suggestion_email_info = (
+            suggestion_services
+            .create_reviewable_suggestion_email_info_from_suggestion(
+                translation_suggestion))
```

Comment:
Why isn't this setup also in the other tests? I'd expect the enabled/disabled tests to have exactly the same arrangement minus one property that triggers the enabled/disabled state. Otherwise, it seems like the tests aren't actually verifying what they say they are.

Reply:
Unless I'm missing something, it still only seems to be here.

Reply:
Is this resolved @kaylahardie?

Reply:
Yeah - I couldn't find the comment thread when I was doing my review but I think it's resolved.